### PR TITLE
More work on things

### PR DESCRIPTION
**Pull request by Machine-Maker**
### Included
* API key creation & authentication
* User login via HangarAuth (using md's fork and branch)

### CODE REVIEW

File: src/main/java/me/minidigger/hangar/controller/api/AuthenticateApiController.java [hunk 1]
```diff
@@ -24,18 +23,21 @@ public AuthenticateApiController(AuthenticationService service) {
...
-            if (authentication instanceof HangarAuthentication) {
-                return ResponseEntity.ok(service.authenticateKeyPublic(body, ((HangarAuthentication) authentication).getUserId()));
-            } else if (authentication.getPrincipal().equals("anonymousUser")) {
-                return ResponseEntity.ok(service.authenticatePublic());
-            } else {
-                throw AuthUtils.unAuth();
-            }
+            return ResponseEntity.ok(service.authenticateKeyPublic(body == null ? null : body.getExpiresIn()));
         }
+            // TODO not sure if all this is needed because you don't need the hangarauth user id, all that is from the api key
```
Comment (MiniDigger) [inline src/main/java/me/minidigger/hangar/controller/api/AuthenticateApiController.java:31]: I think I created this cause I had issue with ano login, if authenticatePublic doesnt use the id, just remoce all of this

File: src/main/java/me/minidigger/hangar/security/HangarAuthentication.java [hunk 2]
```diff
@@ -11,21 +12,30 @@
...
+        this.userId = null;
         setAuthenticated(false);
     }
 
-    public HangarAuthentication(String username, UsersTable usersTable, Collection<GrantedAuthority> authorities) {
+    public HangarAuthentication(Collection<? extends GrantedAuthority> authorities, String username, long userId, UsersTable table) {
         super(authorities);
         this.username = username;
-        this.table = usersTable;
-        super.setAuthenticated(true); // must use super, as we override
```
Comment (MiniDigger) [inline src/main/java/me/minidigger/hangar/security/HangarAuthentication.java:26]: and reason you don't set this to authenticated? does spring even accept it if its not true?

Comment (MiniDigger) [reply src/main/java/me/minidigger/hangar/security/HangarAuthentication.java:26]: actually, the auth manager does this, right? I added the auth manager after I wrote this code, so it should be fine

Comment (Machine-Maker) [reply src/main/java/me/minidigger/hangar/security/HangarAuthentication.java:26]: actually, I'm not sure... other implementations of AbstractAuthenticationToken do call super.setAuthenticated in their constructors... I wasn't seeing any issues, but I'll add it back.

**Pull request merged**